galley/pkg/server/components/validation.go

```diff
@@ -27,10 +27,12 @@ func NewValidation(kubeConfig string, params *validation.WebhookParameters, live
 	return process.ComponentFromFns(
 		// start
 		func() error {
+			webhookServerReady := make(chan struct{})
+			stopCh := make(chan struct{})
 			if params.EnableValidation {
```

Comment:
Would it make sense to do this check in *server.go* instead? Parallel to how we check ```a.EnableServer``` before adding the server component, should we check ```a.ValidationArgs.EnableValidation``` before adding the validation component?

Reply:
@sushicw Thanks for your comments. Right now, I think we cannot do this check in server.go just like what we have done to check `a.EnableServer` before adding the server component. The reason is `ReconcileWebhookConfiguration` can be able to run even if `a.ValidationArgs.EnableValidation` is false. Maybe I can have another PR after this PR to separate the `ReconcileWebhookConfiguration` as a component as well. What do you think?

Reply:
> The reason is ReconcileWebhookConfiguration can be able to run even if a.ValidationArgs.EnableValidation is false. 

I'm not 100% convinced that this is a useful property: when would it make sense to reconcile webhook config but have validation disabled? Furthermore, since the default is to *not* disable ReconcileWebhookConfiguration, aren't we setting users up for unexpected bugs if they don't know about the new flag? It seems like we'd go from needing to change 1 flag to needing to change 2 flags. 

My own instinct is to treat ReconcileWebhookConfiguration as a refinement of EnableValidation, and to force reconcile to be off if EnableValidation is false.

Is there a case where it makes sense to have EnableValidation off and ReconcileWebhookConfiguration on?

Reply:
@sushicw, I tried to describe the case in a [comment thread above](https://github.com/istio/istio/pull/10651#discussion_r246470832).

Reply:
Got it, that makes sense. /lgtm

Reply:
Thanks @ayj  and @sushicw who can help to remove the `do-not-merge` flag if no more comments? Thanks.

Reply:
@ayj I think you originally put the hold on the PR. Are you ok with removing it